content/blog/2022-06-07-using-gcp-remotes-in-dvc.md

```diff
@@ -0,0 +1,178 @@
+the credentials you need to authenticate your GCP account with your project. Go
+ahead and download the credentials file and store it somewhere safe.
+
+![gcp_key.png](/uploads/images/2022-06-07/gcp_key.png)
+
+That’s it for setting up your storage bucket and getting the credentials you
+need! Now let’s add DVC to our demo repo and set up the remote.
+
+## Set up a DVC project
+
+First, add DVC as a requirement to your project with the following installation
+command:
+
+`$ pip install 'dvc[gs]'`
+
+Then you can initialize DVC in your own project with the following command:
+
+`$ dvc init`
+
+This will add all of the DVC internals needed to start versioning your data and
+tracking experiments. Now we need to set up the remote to connect our project
+data stored in GCP to the DVC repo.
+
+### Create a default remote
+
+Now we can make the GCP storage the default remote for the project with the
+following command:
+
+`$ dvc remote add -d bikes gs://updatedbikedata`
+
+This creates a default remote called `bikes` that connects to the
+`updatedbikedata` bucket we made earlier which is where the any data for the
+model will be stored.
+
+### Add GCP credentials
+
+In order for DVC to be able to push and pull data from the remote, you need to
+have valid GCP credentials.
+
+If you are using the GCP CLI already, you should be able to run
+`gcloud auth application-default login`. You can also authenticate with the
+service account we created earlier in a couple of ways with that credentials
+file we downloaded.
+
+You can set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point
+to the credentials file, like this:
+
+```dvc
+$ export GOOGLE_APPLICATION_CREDENTIALS='../tonal-history-154018-e62a79baf90f.json'
+```
+
+Or you can add the credentials file location with the following command:
+
+```dvc
+$ dvc remote modify --local bikes credentialpath '../tonal-history-154018-e62a79baf90f.json'
+```
+
+You can check out more about authentication
+[here in the GCP docs](https://cloud.google.com/sdk/docs/authorizing).
+
+### Push and pull data with DVC
+
+Now you can push data from your local machine to the GCP remote! First, add the
+data you want DVC to track with the following command:
+
+`$ dvc add data`
+
+This will allow DVC to track the entire `data` directory so it will note when
+any changes are made. Then you can push that data to your GCP remote with this
+command:
+
+`$ dvc push`
+
+Here's what that data will look like when it has been successfully uploaded to
+GCP.
+
+![data in GCP](/uploads/images/2022-06-07/data_in_gcp.png)
+
+Then if you move to a different machine or someone else needs to use that data,
+it can be accessed by cloning or forking the project repo, connecting to the
+remote and running:
+
+`$ dvc pull`
+
+That’s it! Now you can connect any DVC project to a GCP storage bucket. If you
+run into any issues, makes sure to check that your credentials are valid, check
```

Comment:
```suggestion
run into any issues, make sure to check that your credentials are valid, check
```
:+1:

It could be worth it to add a section at the top, right after the authentication section, to test out if authentication to the bucket actually works.
For example, when the google cloud SDK is availble, it's enough to run:
```bash
echo foo > testfile
gsutil cp testfile gs://updatedbikedata/testfile
```